pkg/kubectl/cmd/util/openapi/validation/validation.go

```diff
@@ -82,7 +82,7 @@ func (v *SchemaValidation) validateResource(obj interface{}, gvk schema.GroupVer
 		return nil
 	}
 
-	resource := v.resources.LookupResource(gvk)
+	resource := v.resources.LookupResource(gvk.String())
```

Comment:
why are we regressing this to be unstructured? stringly typed data is a prominent source of bugs related to group/version/kind

Reply:
Yeah, I've made a similar comment on a related pull-request. @mengqiy has created #51823 to follow-up.